### PR TITLE
Hide comment input form if user is not logged in.

### PR DESCRIPTION
**Pull request by amuamushu**
With this change, users have to be logged in to add a comment. Otherwise, they are directed to log in.

Comment form after logging in: https://screenshot.googleplex.com/5AGrO4HFUB9.png
Comment form before logging in: https://screenshot.googleplex.com/7MAaOkUmVJy.png

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class HomeServlet extends HttpServlet {
+  private static final String HOME = "/";
```
Comment (billie-cat) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java:13]: Can you make the variable name more descriptive? I do not know what HOME means in terms of "/". Is it HOME_PATH, ...

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = HOME;
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
+      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    } else {
```
Comment (billie-cat) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java:27]: Here is a tip. You want to minimize use of if/else statements as it adds extra unnecessary logic. What you can do is in your if statement, return at the end of it. And then you no longer need an else statement. The logic in there can be outside of any statement. I hope that makes sense.

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java (hunk 1 above)
Comment (amuamushu) [reply portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: Yes, it is referring to the home page.

Done!

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java (hunk 2 above)
Comment (amuamushu) [reply portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: Yes, that makes sense. Thanks for the tip!

Done!

**Pull request merged**